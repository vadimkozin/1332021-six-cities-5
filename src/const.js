export const CITIES = {
  Paris: [48.856614, 2.3522219],
  Cologne: [50.937531, 6.960278600000038],
  Brussels: [50.8503396, 4.3517103],
  Amsterdam: [52.3702157, 4.8951679],
  Hamburg: [53.5510846, 9.99368179999999],
  Dusseldorf: [51.2277411, 6.773455600000034],
};
export const CITIES_NAMES = Object.keys(CITIES);

export const MONTHS = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

export const RATING_STARS_MAX = 5;

export const OFFER_PICTURE_MAX = 6;

export const REVIEW_OUTPUT_MAX = 10;

export const MAP_ZOOM_DEFAULT = 12;

export const HOTELS_NEARBY_MAX = 3;

export const CommentSettings = {
  TEXT_LENGTH_MIN: 50,
  TEXT_LENGTH_MAX: 300,
};

export const TypesHousing = {
  APARTMENT: {name: `apartment`, view: `Apartment`},
  ROOM: {name: `room`, view: `Private Room`},
  HOUSE: {name: `house`, view: `House`},
  HOTEL: {name: `hotel`, view: `Hotel`}
};

export const OfferCardType = {
  CITY_PLACE: 1,
  NEAR_PLACE: 2,
};

export const MapType = {
  VERTICAL: 1,
  HORIZONTAL: 2,
};

export const SortingType = {
  POPULAR: `popular`, // в порядке, полученном с сервера
  LOW_TO_HIGH: `to-high`, // От дешёвых к дорогим.
  HIGH_TO_LOW: `to-low`, // От дорогих к дешёвым.
  TOP_RATED_FIRST: `top-rated` // От высокого рейтинга к низкому
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER_ID: `/offer/:id`,
};

export const APIRoute = {
  OFFERS: `/hotels`,
  LOGIN: `/login`,
  OFFER_ID: `/hotels/`,
};

export const StatusType = {
  IDLE: `idle`,
  LOADING: `loading`,
  SUCCESS: `succes`,
  FAILED: `failed`,
};
