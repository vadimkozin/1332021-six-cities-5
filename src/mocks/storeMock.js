import configureStore from 'redux-mock-store';
import {SortingType, CITIES_NAMES, StatusType, AuthorizationStatus} from '../const';
import {offerAdapter, commentAdapter} from '../store/adapters';

import hotelJson from './json/hotel';
import hotelsNearbyJson from './json/hotels-nearby';
import offersJson from './json/offers';
import commentsJson from './json/comments';

const hotel = offerAdapter.adaptToClientOffer(hotelJson);
const hotelsNearby = offerAdapter.adaptToClient(hotelsNearbyJson);
const offers = offerAdapter.adaptToClient(offersJson);
const comments = commentAdapter.adaptToClient(commentsJson);

const initCity = CITIES_NAMES[0];

const Data = {
  SORT: `sort`,
  CITY: `city`,
  OFFERS: `offersGist`,
  HOTEL: `hotelGist`,
  HOTELS_NEARBY: `hotelsNearbyGist`,
  COMMENTS: `commentsGist`,
};

const basicState = {
  data: null,
  status: StatusType.IDLE,
  error: ``,
};

const makeInit = (basic = basicState) => (data = {}) => Object.assign({}, basic, {data});

const init = makeInit();

const storeConfig = configureStore();

const storeMock = storeConfig({
  data: {
    cities: CITIES_NAMES,
  },
  process: {
    [Data.CITY]: initCity,
    [Data.SORT]: SortingType.POPULAR,
    [Data.OFFERS]: init(offers),
    [Data.HOTEL]: init(hotel),
    [Data.HOTELS_NEARBY]: init(hotelsNearby),
    [Data.COMMENTS]: init(comments),
  },
  user: {
    id: 1,
    name: `qwerty`,
    email: `qwerty@mail.ru`,
    isPro: false,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  },
});

export {storeMock};
