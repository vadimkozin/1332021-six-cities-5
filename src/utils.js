import {MONTHS, RATING_STAR_MAX, TypesHousing} from './const';

// export const formatDate = (date) => `${MONTHS[date.getMonth()]} ${date.getDate()}`;

// добавляет ведущие нули: ( '2' => '02')
const addZeros = (number, digitsInNumber = 2) => {
  let n = String(number);
  while (n.length < digitsInNumber) {
    n = `0` + n;
  }
  return n;
};

// форматривание дат: my:'October 2020' ymd:'2020-10-25'
export const formatDate = {
  my: (date) => `${MONTHS[date.getMonth()]} ${date.getFullYear()}`,
  ymd: (date) => `${date.getFullYear()}-${addZeros(date.getMonth() + 1)}-${addZeros(date.getDate())}`,
};

export const getRatingProc = (rating) => Math.round(rating / RATING_STAR_MAX * 100);

export const getHousingView = (name) => {
  const keys = Object.keys(TypesHousing);
  const type = keys.find((key) => TypesHousing[key].name === name);
  return TypesHousing[type].view;
};

export class UniqId {
  constructor(start = 0) {
    this.start = start;
  }
  get next() {
    return this.start++;
  }
}


