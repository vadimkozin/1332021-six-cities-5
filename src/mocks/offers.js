import {AvatarGenerator} from 'random-avatar-generator';
import {TOWNS, TypesHousing} from '../const';
import {getUUIDGenerator, uniqArray} from '../utils';

const TITLES = [
  `Beautiful studio at great location`,
  `Excellent location`,
  `Amazing view from the window`,
  `Gorgeous view of the sunrise`,
  `Everything you need is available in 5 minutes`,
  `Everything you need`,
];

export const DESCRIPTIONS = [
  `Lorem ipsum dolor sit amet consectetur, adipisicing elit. Voluptates blanditiis dicta ad non porro dignissimos officia asperiores vel repellendus earum, sapiente enim cum cupiditate, soluta possimus. Quidem at aliquam nihil?`,
  `Lorem ipsum dolor sit amet consectetur adipisicing elit. Hic, cum?`,
  `Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta voluptates, velit minima repellat praesentium commodi ea fugiat? Reprehenderit, architecto sint.`,
  `Lorem ipsum dolor sit amet consectetur adipisicing elit. Corrupti facilis maxime adipisci sed voluptate ipsum!`,
  `Lorem ipsum dolor sit amet, consectetur adipisicing elit. `,
  `Nostrum officiis laborum est debitis facilis nulla nihil laboriosam, sapiente rerum nesciunt eveniet.`,
  `Laudantium facere, totam eligendi aspernatur corrupti dolorem suscipit numquam, vero similique.`,
  `Perspiciatis distinctio eum. Aut veritatis ipsum provident tempore?`,
  `Fugiat quia rerum sed minima iusto autem ut maxime molestias libero odio natus temporibus labore tempore magni officiis cumque nemo expedita itaque, possimus et praesentium fuga dicta, pariatur ex? Necessitatibus cum, ipsam quam voluptate, aut perferendis quaerat recusandae amet accusantium assumenda perspiciatis numquam quisquam, animi exercitationem provident fugiat nisi.`,
  `Culpa aliquam cumque modi labore quae est ad illum nobis molestias.`,
];

const HOUSEHOLD_ITEMS = [
  `WiFi`,
  `Heating`,
  `Kitchen`,
  `Cable TV`,
  `Washing machine`,
  `Coffee machine`,
  `Dishwasher`,
  `Towels`,
  `Baby seat`,
  `Electric kettle`,
  `Slippers`,
];

export const NAMES = [
  `ivan`,
  `petr`,
  `alex`,
  `angela`,
  `maksim`,
  `vadim`,
  `anton`,
  `irina`,
  `inna`,
  `oksana`,
];

const URL_PHOTO = `http://picsum.photos/260/200?r=`;

const generator = new AvatarGenerator();

const getRandomPhoto = () => `${URL_PHOTO}${Math.random()}`;

// возвращает случайное число из диапазона между min и max (оба включены)
export const getRandomInteger = (min = 0, max = 0) => Math.floor(Math.random() * (max - min + 1)) + min;

export const getRandomFloat = (min = 0, max = 0, fixed = 2) => Number(((Math.random() * (max - min)) + min).toFixed(fixed));

export const getRandomFrom = (array) => {
  const index = getRandomInteger(0, array.length - 1);
  return array[index];
};

export const getRandomDate = (daysBeforeMax = 31) => {
  const hours = getRandomInteger(1, 24 * daysBeforeMax);

  return new Date(Date.now() - 1000 * 3600 * hours);
};

const getRandomPartArray = (array) => {
  const itemsCount = getRandomInteger(0, array.length - 1);

  return uniqArray(Array(itemsCount).fill()
    .map(() => array[getRandomInteger(0, array.length - 1)]));
};

const getRandomType = () => {
  const keys = Object.keys(TypesHousing);
  const index = getRandomInteger(0, keys.length - 1);

  return TypesHousing[keys[index]].name;
};

const getRandomPictures = (from = 1, to = 5) => {
  const count = getRandomInteger(from, to);

  return Array(count).fill().map((it, i) => ({
    src: getRandomPhoto(),
    alt: `Place image-${i}`,
  }));

};

const getRandomOwner = () => {
  return {
    avatar: generator.generateRandomAvatar(),
    name: getRandomFrom(NAMES),
    isSuper: Boolean(getRandomInteger(0, 1)),
  };
};

const uuid = getUUIDGenerator();

const getOffers = (count) => {
  return Array(count).fill().map(() => ({
    id: uuid(),
    pictures: getRandomPictures(3, 10),
    title: getRandomFrom(TITLES),
    description: getRandomFrom(DESCRIPTIONS),
    isPremium: Boolean(getRandomInteger(0, 1)),
    typeHousing: getRandomType(),
    rating: getRandomFloat(1, 5),
    bedroomsNumber: getRandomInteger(1, 5),
    guestsMax: getRandomInteger(1, 3),
    price: getRandomInteger(50, 700),
    householdItems: getRandomPartArray(HOUSEHOLD_ITEMS),
    owner: getRandomOwner(),
    town: getRandomFrom(TOWNS),
    isFavorite: Boolean(getRandomInteger(0, 1)),
  }));
};

const offers = getOffers(5);

export default offers;

