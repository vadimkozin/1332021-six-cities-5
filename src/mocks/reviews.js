import {AvatarGenerator} from 'random-avatar-generator';
import {NAMES, DESCRIPTIONS, getRandomFrom, getRandomFloat, getRandomDate, getRandomInteger} from '../mocks/offers';
import {getUUIDGenerator} from '../utils';

const generator = new AvatarGenerator();

const uuid = getUUIDGenerator();

const getReviews = (count) => {
  return Array(count).fill().map(() => ({
    id: uuid(),
    text: getRandomFrom(DESCRIPTIONS),
    date: getRandomDate(),
    rating: getRandomFloat(1, 5),
    user: {
      id: uuid(),
      avatarUrl: generator.generateRandomAvatar(),
      name: getRandomFrom(NAMES),
      isPro: Boolean(getRandomInteger(0, 1)),
    },
  }));
};
export default getReviews(3);
