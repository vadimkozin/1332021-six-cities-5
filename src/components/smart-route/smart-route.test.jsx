import React from 'react';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import {render} from 'enzyme';
import storeMock from '../../mocks/storeMock';
import SmartRoute from './smart-route';
import {noop} from '../../mocks/utils';
import {AppRoute} from '../../const';

describe(`RoomPage`, () => {
  it(`should render RoomPage component`, () => {
    const component = render(
        <Provider store={storeMock}>
          <MemoryRouter>
            <SmartRoute render={noop} path={AppRoute.LOGIN} exact redirectTo={AppRoute.ROOT} />
          </MemoryRouter>
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
