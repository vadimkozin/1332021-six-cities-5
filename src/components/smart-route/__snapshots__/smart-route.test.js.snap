// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomPage should render RoomPage component 1`] = `null`;
